### PR TITLE
refactor(character): Use format strings

### PR DESCRIPTION
**Pull request by ATiltedTree**
#### Description

This PR deprecates `style_failure`, `style_success` and `use_symbol_for_status` keys in the `character` module to replace it with the `format` key instead.

#### Motivation and Context

Related to issue #1057

#### How Has This Been Tested?

- [ ] I have tested using **MacOS**
- [x] I have tested using **Linux**
- [ ] I have tested using **Windows**

#### Checklist:

- [x] I have updated the documentation accordingly.
- [x] I have updated the tests accordingly.

### CODE REVIEW

File: tests/testsuite/character.rs [hunk 1]
```diff
@@ -74,77 +77,75 @@ fn char_module_symbolyes_status() -> io::Result<()> {
...
         .output()?;
     let actual = String::from_utf8(output.stdout).unwrap();
-    assert!(actual.contains(&expected_other));
+    assert_eq!(expected_other, actual);
 
     Ok(())
 }
 
 #[test]
 fn char_module_fish_keymap() -> io::Result<()> {
```
Comment (heyrict) [inline tests/testsuite/character.rs:117]:
```suggestion
fn fish_keymap() -> io::Result<()> {
```
The forgotten one.

File: tests/testsuite/character.rs [hunk 2]
```diff
@@ -74,77 +77,75 @@ fn char_module_symbolyes_status() -> io::Result<()> {
...
 #[test]
-fn char_module_zsh_keymap() -> io::Result<()> {
-    let expected_vicmd = "❮";
-    // TODO make this less... well, stupid when ANSI escapes can be mocked out
-    let expected_specified = "I HIGHLY DOUBT THIS WILL SHOW UP IN OTHER OUTPUT";
-    let expected_other = "❯";
+fn zsh_keymap() -> io::Result<()> {
+    let expected_vicmd = format!("{} ", Color::Green.bold().paint("❮"));
+    let expected_specified = format!("{} ", Color::Green.bold().paint("V"));
+    let expected_other = format!("{} ", Color::Green.bold().paint("❯"));
```
Comment (heyrict) [inline tests/testsuite/character.rs:83]: Looks brilliant :)

**Comment by heyrict** [comment]
Thanks for your fast response! Could you rebase your branch onto the latest feat/format-string branch. I've pushed a fix for the errors in CI.

**Comment by ATiltedTree** [comment]
I'm currently away from home and only on my phone so I can't currently rebase it. I will do it as soon as I'm home!

**Comment by ATiltedTree** [comment]
Rebased 👍

**Pull request merged**